src/txn/txn_rollback_to_stable.c

```diff
@@ -1865,7 +1875,15 @@ __rollback_to_stable_btree_apply_all(WT_SESSION_IMPL *session, wt_timestamp_t ro
     }
     WT_ERR_NOTFOUND_OK(ret, false);
 
-    if (F_ISSET(S2C(session), WT_CONN_RECOVERING))
+    /*
+     * Performing eviction in parallel to a checkpoint can lead to situation where the history store
+     * have more updates than its corresponding data store. Performing history store cleanup at the
+     * end can able to remove any such unstable updates that are written to the history store.
+     *
+     * Do not perform the final pass on the history store in an in-memory configuration as it
+     * doesn't exist.
+     */
+    if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
```

Comment:
So this is necessary to run in the live database case? My understanding is that we can still skip this for live database

Reply:
We need to run in all the scenarios to clean up any stale entries present in the history store. The actual test format problem occurred running rollback to stable in a live system followed by the update operation.